frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -207,9 +207,9 @@ export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModal
       };
       const patch = [
         {
-          op: 'add',
-          path: '/metadata/labels/thislabel',
-          value: `schedule-${pvcLabel}`,
+          op: 'replace',
+          path: '/metadata/labels',
+          value: { ...resource.metadata.labels, [snapshotName]: `schedule-${pvcLabel}` },
```

Comment:
You should check `op: resource.metadata.labels ? 'replace' : 'add' ` before deciding the operand. Each operand has its own meaning.